draft-ietf-quic-qpack.md

```diff
@@ -619,14 +619,22 @@ QPACK defines two unidirectional stream types:
    It carries an unframed sequence of decoder instructions from decoder
    to encoder.
 
-<!-- s/exactly/no more than/  ? -->
 HTTP/3 endpoints contain a QPACK encoder and decoder. Each endpoint MUST
-initiate a single encoder stream and decoder stream. Receipt of a second
-instance of either stream type be MUST treated as a connection error of type
+initiate at most one encoder stream and one decoder stream. Receipt of a second
+instance of either stream type MUST be treated as a connection error of type
 HTTP_WRONG_STREAM_COUNT. These streams MUST NOT be closed. Closure of either
 unidirectional stream type MUST be treated as a connection error of type
 HTTP_CLOSED_CRITICAL_STREAM.
 
+An endpoint MAY avoid creating its own encoder stream if the maximum size of
```

Comment:
If I don't want to use an encoder stream, I still have to open it if the table size is non-zero?  I thought the intent was to be permissive of not opening, so this seems unintentionally restrictive.

Reply:
I think this is telling that the current definition of signalling around QPACK is all a bit wrong. The receiver tells the sender if it is willing to support dynamic compression but there is no explicit signal it will happen until a stream is opened, are we happy with that?

For @ianswett's issue, an additional clause to this sentence could help e.g. 

"An endpoint MAY avoid creating its own encoder stream if the maximum size of the dynamic table permitted by the peer is zero, or if the endpoint doesn't want to use the dynamic table."

Reply:
@LPardue : The explicit signal is the arrival of a Set Dynamic Table Capacity instruction on the encoder stream with a non-zero capacity (or perhaps the arrival of a header block with a non-zero Required Insert Count, if there's reordering or loss).  I don't see any problem with this - what's the specific concern?

It's actually an improvement over HPACK, where the receiver assumes the sender will use the table by virtue of the non-zero default max capacity.  Implementations that never use the dynamic table might not be kind enough to send an Update Table Size with size=0.

For the language here, would it read better the other way around: 

"The encoder need only open the encoder stream if it wishes to use the dynamic table and is permitted to do so by the decoder" - or something to that effect?